plugins/devlink/fu-devlink-component.c

```diff
@@ -160,12 +157,9 @@ fu_devlink_component_build_instance_id(FuDevice *device,
 	};
 	g_auto(GStrv) keys = NULL;
 
-	if (driver_name != NULL) {
-		fu_device_add_instance_str(device, "DRIVER", driver_name);
-		g_strv_builder_add(keys_builder, "DRIVER");
-	}
```

Comment:
The driver is one of the keys in devlink-based guids. I'm not sure why you remove it here.

Reply:
Putting it in the instance ID it becomes API -- so even if the kernel driver is renamed then all the existing firmware would not be installable. i.e. we only really export "visible" instance IDs where we _or the device_ controls the value.

Reply:
> Putting it in the instance ID it becomes API -- so even if the kernel driver is renamed then all the existing firmware would not be installable. i.e. we only really export "visible" instance IDs where we _or the device_ controls the value.

If the vendor renames a driver (honestly, I don't recall that happening in networking area for last 15 years), it is his responsibility to do preparations in advance and upload appropriate guids to LVFS. LVFS updates will not be the only thing broken.

Reply:
Would there ever be more than one DRIVER value for a specific VID/PID?